numpyro/infer/hmc.py

```diff
@@ -401,12 +401,13 @@ def __init__(self,
         self._model = model
         self._potential_fn = potential_fn
         self._kinetic_fn = kinetic_fn if kinetic_fn is not None else euclidean_kinetic_energy
-        self._step_size = step_size
+        self._step_size = float(step_size) if isinstance(step_size, int) else step_size
```

Comment:
this resolves issues when users provide wrong dtypes such as `step_size=1` or `trajectory_length=6`.